ros_buildfarm/templates/doc/doc_independent_docker_job.xml.em

```diff
@@ -81,14 +87,18 @@
 @{
 import os
 doc_repository_name = os.path.splitext(os.path.basename(doc_repository_url))[0]
+
+if doc_repository_branch is None:
+  doc_repository_branch = ''
+else:
+  doc_repository_branch = '--no-single-branch -b ' + doc_repository_branch
 }@
 @(SNIPPET(
     'builder_shell',
     script='\n'.join([
         'echo "# BEGIN SECTION: Clone %s"' % doc_repository_name,
-        'python3 -u $WORKSPACE/ros_buildfarm/scripts/wrapper/git.py clone --depth 1 %s $WORKSPACE/repositories/%s' % (doc_repository_url, doc_repository_name),
+        'python3 -u $WORKSPACE/ros_buildfarm/scripts/wrapper/git.py clone --depth 1 %s %s $WORKSPACE/repositories/%s' % (doc_repository_url, doc_repository_branch, doc_repository_name),
         'git -C $WORKSPACE/repositories/%s log -n 1' % doc_repository_name,
-        'rm -fr $WORKSPACE/repositories/%s/.git' % doc_repository_name,
```

Comment:
This has been there since the doc_independent_docker_job was added.  I can only guess that this was a copy from the doc_independent_job, which does the same thing.  Maybe it is an attempt to limit the number of Docker layers?  I don't really know.

Reply:
It does occur to me that some of the tools which recurse over projects may stray into .git metadata if left unchecked but I'm okay giving this a shot and monitoring in production. https://github.com/ros-infrastructure/ros_buildfarm/commit/c204239b08fdec35c2e6237fdde4bba7504f382e is the commit which introduced this for the doc_independent job and appears to be part of a generic cleanup approach. The commit message does not indicate a motivation.

Reply:
That's true, I think we'll have to monitor it.  The good news is that this job is used for very few jobs currently (only index at the moment, I believe), so any breakage will be limited in scope.